.github/workflows/publish-doc-site.yml

```diff
@@ -0,0 +1,30 @@
+# This is a basic workflow to help you get started with Actions
+
+name: publish doc site
+
+# Controls when the action will run. 
+on:
+  # Triggers the workflow on push or pull request events but only for the master branch
+  push:
+    branches: [ master ]
```

Comment:
I think this is saying it will run when we push branches to master... 🤞

Reply:
I think that's correct!

[question] do we currently deploy the documentation site on every merge to master? Does it ever lead to a situation where the documentation site is updated with the latest code in master but master hasn't been published as a new NPM version yet?

Reply:
`[question] do we currently deploy the documentation site on every merge to master?` - No, we want to deploy doc site only when there's a release to NPM so that the version on NPM & doc site should be the same

Reply:
So I guess I should remove the `push` section of this action then?